### ISSUE TITLE
Intro message time too short to read

### ISSUE THREAD

**Issue by ianks** [neovim/neovim#3416]
I have noticed that the intro message is a bit too short to read. I was looking through the code to try and find where this timeout would be set, but came up empty handed. Can someone push me in the right direction so I can make my first Neovim contribution? :smile:

**Comment by justinmk** [comment]
Either version.c or messages.c.

**Comment by fdinoff** [comment]
I can't reproduce. Mine stays until I edit the buffer. Can you confirm that this happens with `nvim -u NONE`? 

(I'm currently running 6bcbb18dc88d9b84f71c09450af10f8f537a8a68 which is about 2 weeks old at the moment.)

**Comment by fdinoff** [comment]
I just updated to f454e4bb11a7a69f1c03325016e3b95cda2fc08e, still can't reproduce.

**Comment by justinmk** [comment]
Pretty sure it is random.

**Comment by fdinoff** [comment]
> Pretty sure it is random. 

Mine pretty consistently stays for a long time. (As in I could probably walk away and it would never disappear)

> Or a shortmess option.

The only shortness option I see is `I` which disables it. (There is the `:intro` command to show the intro)

**Comment by justinmk** [comment]
See https://github.com/neovim/neovim/pull/1885#issuecomment-117128528

On second thought, I suspect it may be a redraw or event loop cycle. See https://github.com/neovim/neovim/issues/3382 and https://github.com/neovim/neovim/pull/3413

@ianks Does it happen with `nvim -u NONE`?

**Comment by fwalch** [comment]
I think deoplete caused the intro screen to disappear after a short time, not sure if that's still the case.

**Comment by justinmk** [comment]
The general problem here is that job callbacks or remote requests can cause a redraw. This may just a holdover from assumptions the `main_loop` logic. E.g. maybe we could set `skip_redraw` when handling a job or request. cc @tarruda

**Comment by tarruda** [comment]
> The general problem here is that job callbacks or remote requests can cause a redraw. This may just a holdover from assumptions the main_loop logic. E.g. maybe we could set skip_redraw when handling a job or request. cc @tarruda

Yes. Since events return to the main loop as `K_EVENT`, it hits redraw logic as if any other key were pressed. I bet it is the same problem described by @fwalch: Remote plugins can result in redraws at startup. Even though I'm trying to get rid of `K_EVENT` in #3413, we still have to execute redraw checks to ensure asynchronous calls are visible by the user.

I have a possible solution in mind but it will only probably be achieved in 0.2 or 0.3.

**Comment by justinmk** [comment]
See https://github.com/neovim/neovim/pull/3413#issuecomment-151023624 for repro steps and further discussion.

**Comment by vito-c** [comment]
+1 this bug causes the completion menu to be unusable when there is a process running in the background.

**Comment by justinmk** [comment]
Redraws caused by `:terminal` activity and FocusGained/FocusLost were fixed orthogonally.

The [remaining behavior](https://github.com/neovim/neovim/issues/3416#issuecomment-145282126) is not a bug but it would be nice to think about alternatives.

Moved to `unplanned` milestone and removed `bug` tag.
